src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -160,14 +166,61 @@ object DepositHandler {
     }
   }
 
-  private def checkBagValidity(bagitDir: File)(implicit id: String): Try[Unit] = {
+  def resolveFetchItems(bagitDir: File)(implicit id: String): Try[Unit] = Try {
+    log.debug(s"[$id] Checking if bag contains fetch.txt")
+    getFetchTxt(bagitDir).foreach(fetchText =>
+      { checkFetchTxtUrls(fetchText)
+        resolveFetchTxtFiles(bagitDir, fetchText)
+      }
+    )
+  }
+
+  private def getFetchTxt(bagitDir: File)(implicit id: String): Option[FetchTxt] = Option {
+    getBagFromDir(bagitDir).getFetchTxt
+  }
+
+  private def checkFetchTxtUrls(fetchText: FetchTxt)(implicit id: String): Unit = {
+    log.debug(s"[$id] Checking validity of urls in fetch.txt")
+    fetchText.asScala.foreach(item =>
+      checkUrlValidity(item.getUrl))
+  }
+
+  private def checkUrlValidity(url: String)(implicit id: String): Unit = {
+    // check if the url is syntactically correct
+    try {
+      new URL(url)
+    } catch {
```

Comment:
You are right. I have now `MalformedURLException`, which catches syntactically wrong urls, including `null`.  All the other exceptions will be thrown higher up as `Unexpected failure in deposit` with the original cause.